src/NumericalAlgorithms/Interpolation/PredictedZeroCrossing.cpp

```diff
@@ -16,4 +16,15 @@ double predicted_zero_crossing_value(const std::vector<double>& x_values,
   return -coefficients[0]/coefficients[1];
 }
 
+std::vector<double> predicted_zero_crossing_value(
+    const DataVector& x_values, const std::vector<DataVector>& y_values) {
```

Comment:
I am thinking of the primary use case for this function:
x_values will be a set of times.  y_values will be a set of DataVectors, one DataVector for each time, where each DataVector is the characteristic speed evaluated at all points on the surface.
Then the result will be the zero-crossing time for each point on the surface.

So I think x_values be a `std::vector<double>`.
Then there could be a check that `x_values.size()==y_values.size()`.  

And then I think the return type should be `DataVector` (with the same size as the DataVector in y_values).

Reply:
I'm using @gsb76 's `LinearLeastSquares` function that takes `LinearLeastSquares<Order>::fit_coefficients(const T& x_values, const std::vector<T>& y_values)` for the arguments to do the fitting for each `std::vector<DataVector` corresponding to the `y_values`. Would it be okay to leave as is and just add the check for `x_values.size()==y_values.size()`  and generalize the return type to be just `DataVector` ? @markscheel

Reply:
I don't think the current implementation of this function will handle the use case I am thinking of.
This is for two reasons: One is that we don't want the user to be copying things into different structures in order to call the function.  But the second reason is because I think you can't just call `LinearLeastSquares` in the same way you do now.  Here's why:

The user (in the use cases I am thinking of) will have a `std::vector<double>` or maybe even a `std::deque<double>` for the x_values.  This will represent a set of times.

The user will also have a `std::vector<DataVector>` or maybe a `std::deque<DataVector>` for the y_values, where each DataVector is the characteristic speed at all the points on a surface.  The y_values are indexed by `yvalues[x_value_index][datavector_index]`.   *This is the opposite indexing order compared to what `LinearLeastSquares` expects for its y_values.*